api/v1alpha1/validation.go

```diff
@@ -681,3 +754,39 @@ func validateGraceDuration(schedule cron.Schedule, duration string) error {
 
 	return nil
 }
+
+// To ensure valid go-template syntax, we create a fake device that contains only ObjectMeta
+// (what we allow to include in parameters), then evaluate.
+type whitelistDevice struct {
+	Metadata ObjectMeta
+}
+
+func ValidateParametersInString(s *string, path string) (bool, []error) {
+	if s == nil {
+		return false, []error{}
+	}
+
+	allErrs := []error{}
+
+	t, err := template.New("t").Parse(*s)
```

Comment:
How expensive is validating with templates vs. using pre-compiled regexes here?

Reply:
We can use a regex here if we want to limit the use of templates to a very small